### ISSUE TITLE
No alternative to Twitter login

### ISSUE THREAD

**Issue by coggy9** [MehediH/winstall#138]
Twitter is [removing free API access](https://twitter.com/TwitterDev/status/1621026986784337922) on the 9th, and Winstall only has Twitter as a login option. There needs to be either another method of logging in, or the ability to create/save packs without an account.

**Comment by funky123** [comment]
yes, add login with github instead

**Comment by mikhoul** [comment]
> yes, add login with github instead

@MehediH 
I fully agree, we should be able to login with our email or Github at least.

Regards :octocat:

**Comment by Gitlish** [comment]
Email login, please.

**Comment by radonn24** [comment]
Bump this topic/issue ...

@MehediH - Sorry for the random / late ping ... I figured you might be the best person to ask on this ...

Haven't checked the site in a while, and went to grab a Powershell command for an app I had listed in a "pack" I created, and noticed a Twitter API issue/message restricting functionality. I'm a "novice" web developer (at best), but it got me thinking about possible solutions, and - despite accepting this is most likely inaccurate - on the small chance this *might* help in any way, I wanted to return the favor as your site has helped also. Here's my thoughts ...

(Before the recent Twitter API v2 changes) Upon user account registration/validation via the winstall.app website login oAuth link, some basic permission requests for simple account information are prompted ...

![image](https://github.com/MehediH/winstall/assets/13595375/013b0971-d0af-4aad-994e-f9eec43b8257)

Was the user's email address returned via the requested permission(s)? (It's been a number of years since I have used Twitter's oAuth, but I think [this might be the docs](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/manage-account-settings/api-reference/get-account-verify_credentials) for this).

... If so, was the user(s) email address hashed, then databased for future account validation/reference? ... If so, would swapping for any other 3rd-party oAuth provider be feasible, using the previously hashed email address as reference to avoid account collision(s) and/or merge multiple oAuth providers without creating duplicate accounts?

Hopefully, this makes some sense, even if this is not the way the site validates users.

No explanations or replies needed, or expected. Maybe this might help somehow?? Maybe not? Not sure. Just wanted to share my thoughts. Thank you for any/all your efforts already! 

radonn

**Comment by link6155** [comment]
Twitter/X login currently does not work. Trying click login leads to the following:
![image](https://github.com/omaha-consulting/winstall/assets/1360980/5e4262d9-53b3-4257-8be0-c6f70c153fb6)

Tested on Firefox and Edge

**Comment by bepstein111** [comment]
Is there a timeline to get this fixed at all?

**Comment by st-jliu** [comment]
We will add the auth feature back soon.

**Comment by st-jliu** [comment]
Dear coggy9

This project has been taken over by Splashtop. Login options via Google, GitHub, and Microsoft are now available; therefore, we are closing this ticket. We invite you to try the new solution and provide further feedback or suggestions.
